// @bigcommerce/big-design/dist/es/components/Switch/Switch.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React, { forwardRef, memo, useId } from 'react';
import { Flex } from '../Flex';
import { HiddenCheckbox, StyledSwitchLabel } from './styled';
export const RawSwitch = _ref => {
  let {
    checked,
    disabled,
    forwardedRef,
    ...props
  } = _ref;
  const uniqueSwitchId = useId();
  const id = props.id ? props.id : uniqueSwitchId;
  return /*#__PURE__*/React.createElement(Flex, null, /*#__PURE__*/React.createElement(HiddenCheckbox, _extends({
    checked: checked,
    disabled: disabled,
    id: id,
    type: "checkbox"
  }, props, {
    ref: forwardedRef
  })), /*#__PURE__*/React.createElement(StyledSwitchLabel, {
    "aria-hidden": true,
    checked: checked,
    disabled: disabled,
    htmlFor: id
  }));
};
export const Switch = /*#__PURE__*/memo( /*#__PURE__*/forwardRef((_ref2, ref) => {
  let {
    className,
    style,
    ...props
  } = _ref2;
  return /*#__PURE__*/React.createElement(RawSwitch, _extends({}, props, {
    forwardedRef: ref
  }));
}));
