// @bigcommerce/big-design/dist/cjs/components/InlineMessage/InlineMessage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineMessage = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _bigDesignIcons = require("@bigcommerce/big-design-icons");
var _react = _interopRequireWildcard(require("react"));
var _mixins = require("../../mixins");
var _utils = require("../../utils");
var _Box = require("../Box");
var _Button = require("../Button");
var _private = require("../Button/private");
var _Grid = require("../Grid");
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const defaultLocalization = {
  close: 'Close'
};
const InlineMessage = /*#__PURE__*/(0, _react.memo)(_ref => {
  let {
    className,
    style,
    header,
    localization = defaultLocalization,
    ...props
  } = _ref;
  const filteredProps = (0, _mixins.excludePaddingProps)(props);
  const icon = (0, _react.useMemo)(() => props.type && (0, _utils.getMessagingIcon)(props.type, true), [props.type]);
  const renderedMessages = (0, _react.useMemo)(() => props.messages.map((_ref2, index) => {
    let {
      text,
      link
    } = _ref2;
    return /*#__PURE__*/_react.default.createElement(_Box.Box, {
      key: index
    }, /*#__PURE__*/_react.default.createElement(_styled.StyledMessageItem, null, text), ' ', link && /*#__PURE__*/_react.default.createElement(_styled.StyledLink, link, link.text));
  }), [props.messages]);
  const renderedHeader = (0, _react.useMemo)(() => header && /*#__PURE__*/_react.default.createElement(_styled.StyledHeader, null, header), [header]);
  const renderedActions = (0, _react.useMemo)(() => props.actions && /*#__PURE__*/_react.default.createElement(_styled.StyledActionsWrapper, {
    flexDirection: "row",
    flexWrap: "wrap",
    marginTop: "xSmall"
  }, props.actions.map((_ref3, index) => {
    let {
      text,
      variant = 'secondary',
      ...actionProps
    } = _ref3;
    return /*#__PURE__*/_react.default.createElement(_Button.Button, (0, _extends2.default)({}, (0, _mixins.excludeMarginProps)(actionProps), {
      key: index,
      marginBottom: "xSmall",
      marginHorizontal: "xxSmall",
      mobileWidth: "auto",
      variant: (0, _utils.getActionVariant)(variant)
    }), text);
  })), [props.actions]);
  return /*#__PURE__*/_react.default.createElement(_styled.StyledInlineMessage, (0, _extends2.default)({}, filteredProps, {
    backgroundColor: "white",
    role: "alert"
  }), /*#__PURE__*/_react.default.createElement(_Grid.GridItem, {
    gridArea: "icon"
  }, icon), /*#__PURE__*/_react.default.createElement(_Grid.GridItem, {
    gridArea: "messages"
  }, renderedHeader, renderedMessages, renderedActions), props.onClose && /*#__PURE__*/_react.default.createElement(_Grid.GridItem, null, /*#__PURE__*/_react.default.createElement(_private.MessagingButton, {
    iconOnly: /*#__PURE__*/_react.default.createElement(_bigDesignIcons.CloseIcon, {
      size: "medium",
      title: localization.close
    }),
    onClick: props.onClose
  })));
});
exports.InlineMessage = InlineMessage;
InlineMessage.defaultProps = {
  messages: [],
  type: 'success'
};
