// @bigcommerce/big-design/dist/es/components/Collapse/Collapse.js
import { ExpandMoreIcon } from '@bigcommerce/big-design-icons';
import React, { useId, useState } from 'react';
import { Box } from '../Box';
import { StyledButton } from './styled';
export const Collapse = _ref => {
  let {
    children,
    title,
    onCollapseChange,
    initiallyOpen = false
  } = _ref;
  const [isOpen, setIsOpen] = useState(initiallyOpen);
  const triggerId = useId();
  const panelId = useId();
  const handleTitleClick = () => {
    const nextIsOpen = !isOpen;
    setIsOpen(nextIsOpen);
    if (typeof onCollapseChange === 'function') {
      onCollapseChange(nextIsOpen);
    }
  };
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(StyledButton, {
    "aria-controls": panelId,
    "aria-expanded": isOpen,
    iconRight: /*#__PURE__*/React.createElement(ExpandMoreIcon, {
      title: title
    }),
    id: triggerId,
    isOpen: isOpen,
    marginVertical: "small",
    onClick: handleTitleClick,
    variant: "subtle"
  }, title), /*#__PURE__*/React.createElement(Box, {
    "aria-labelledby": triggerId,
    display: isOpen ? 'block' : 'none',
    hidden: !isOpen,
    id: panelId,
    role: "region"
  }, children));
};
