// @bigcommerce/big-design-theme/dist/es/system/colors.js
export const colors = {
  currentColor: 'currentColor',
  inherit: 'inherit',
  transparent: 'transparent',
  white: '#FFFFFF',
  brand: '#273A8A',
  secondary: '#B4BAD1',
  secondary10: '#F6F7FC',
  secondary20: '#ECEEF5',
  secondary30: '#D9DCE9',
  secondary40: '#B4BAD1',
  secondary50: '#8C93AD',
  secondary60: '#5E637A',
  secondary70: '#313440',
  primary: '#3C64F4',
  primary10: '#F0F3FF',
  primary20: '#DBE3FE',
  primary30: '#9EB3FC',
  primary40: '#3C64F4',
  primary50: '#2852EB',
  primary60: '#0B38D9',
  primary70: '#0024A6',
  danger: '#DB3643',
  danger10: '#FFF0F1',
  danger20: '#FEDBDE',
  danger30: '#FC9EA6',
  danger40: '#DB3643',
  danger50: '#CC1F1F',
  danger60: '#AD0000',
  danger70: '#80000B',
  warning: '#FFBF00',
  warning10: '#FFF9E6',
  warning20: '#FFF0BF',
  warning30: '#FFE180',
  warning40: '#FFBF00',
  warning50: '#FFAE00',
  warning60: '#E58F17',
  warning70: '#CC720A',
  success: '#2AAB3F',
  success10: '#EEFCF0',
  success20: '#D3F5D9',
  success30: '#88D895',
  success40: '#2AAB3F',
  success50: '#208831',
  success60: '#146622',
  success70: '#004D0D'
};
