// @bigcommerce/big-design/dist/es/components/Stepper/Step/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import { hideVisually } from 'polished';
import styled from 'styled-components';
import { Box } from '../../Box';
import { Flex } from '../../Flex';
import { Grid } from '../../Grid';
import { StyleableText } from '../../Typography/private';
export const StyledStep = /*#__PURE__*/styled(Grid).withConfig({
  displayName: "styled__StyledStep",
  componentId: "sc-kstb4o-0"
})(["flex-basis:100%;gap:", ";grid-template-areas:'light dash' 'text text';grid-template-columns:auto 1fr;", "{grid-template-columns:", " 1fr;grid-template-rows:", " 1fr;max-width:", ";min-width:", ";}"], _ref => {
  let {
    theme
  } = _ref;
  return theme.spacing.xSmall;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.breakpoints.tablet;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.xLarge;
}, _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.spacing.xLarge;
}, _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.helpers.remCalc(228);
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.helpers.remCalc(96);
});
export const StyledSrOnlyText = /*#__PURE__*/styled.span.withConfig({
  displayName: "styled__StyledSrOnlyText",
  componentId: "sc-kstb4o-1"
})(["", "{", "}"], _ref7 => {
  let {
    theme
  } = _ref7;
  return theme.breakpoints.tablet;
}, hideVisually());
export const StyledLight = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styled__StyledLight",
  componentId: "sc-kstb4o-2"
})(["border-radius:", ";color:", ";grid-area:light;user-select:none;", "{border-radius:", ";font-size:", ";line-height:", ";}"], _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.spacing.medium;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.colors.white;
}, _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.breakpoints.tablet;
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.borderRadius.circle;
}, _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.typography.fontSize.medium;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.lineHeight.medium;
});
export const StyledDash = /*#__PURE__*/styled(Box).withConfig({
  displayName: "styled__StyledDash",
  componentId: "sc-kstb4o-3"
})(["align-self:center;height:", ";grid-area:dash;width:100%;"], _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.helpers.remCalc(2);
});
export const StyledText = /*#__PURE__*/styled(StyleableText).withConfig({
  displayName: "styled__StyledText",
  componentId: "sc-kstb4o-4"
})(["grid-area:text;"]);
StyledStep.defaultProps = {
  theme: defaultTheme
};
StyledSrOnlyText.defaultProps = {
  theme: defaultTheme
};
StyledLight.defaultProps = {
  theme: defaultTheme
};
StyledDash.defaultProps = {
  theme: defaultTheme
};
StyledText.defaultProps = {
  theme: defaultTheme
};
