// @bigcommerce/big-design/dist/cjs/components/Typography/private.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "StyleableH0", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableH0;
  }
});
Object.defineProperty(exports, "StyleableH1", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableH1;
  }
});
Object.defineProperty(exports, "StyleableH2", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableH2;
  }
});
Object.defineProperty(exports, "StyleableH3", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableH3;
  }
});
Object.defineProperty(exports, "StyleableH4", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableH4;
  }
});
Object.defineProperty(exports, "StyleableHR", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableHR;
  }
});
Object.defineProperty(exports, "StyleableSmall", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableSmall;
  }
});
Object.defineProperty(exports, "StyleableText", {
  enumerable: true,
  get: function () {
    return _Typography.StyleableText;
  }
});
var _Typography = require("./Typography");
