// @bigcommerce/big-design/dist/cjs/components/Typography/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledText = exports.StyledSmall = exports.StyledHR = exports.StyledH4 = exports.StyledH3 = exports.StyledH2 = exports.StyledH1 = exports.StyledH0 = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _polished = require("polished");
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _mixins = require("../../mixins");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const commonTextStyles = props => (0, _styledComponents.css)(["color:", ";margin:0 0 ", ";", ";"], _ref => {
  let {
    theme
  } = _ref;
  return props.color ? theme.colors[props.color] : theme.colors.secondary70;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.spacing.medium;
}, props.ellipsis && (0, _polished.ellipsis)());
const textModifiers = props => (0, _styledComponents.css)(["", " ", " ", " ", " ", " ", " ", ""], _ref3 => {
  let {
    theme
  } = _ref3;
  return props.bold && (0, _styledComponents.css)(["font-weight:", ";"], theme.typography.fontWeight.semiBold);
}, () => props.italic && (0, _styledComponents.css)(["font-style:italic;"]), () => props.underline && (0, _styledComponents.css)(["text-decoration:underline;"]), () => props.strikethrough && (0, _styledComponents.css)(["text-decoration:line-through;"]), () => props.capitalize && (0, _styledComponents.css)(["text-transform:capitalize;"]), () => props.lowercase && (0, _styledComponents.css)(["text-transform:lowercase;"]), () => props.uppercase && (0, _styledComponents.css)(["text-transform:uppercase;"]));
const StyledH0 = /*#__PURE__*/_styledComponents.default.h1.withConfig({
  displayName: "styled__StyledH0",
  componentId: "sc-tqnj75-0"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.typography.fontSize.xxxLarge;
}, _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.typography.fontWeight.extraLight;
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.lineHeight.xxxLarge;
}, _ref7 => {
  let {
    theme
  } = _ref7;
  return theme.spacing.xLarge;
}, (0, _mixins.withMargins)());
exports.StyledH0 = StyledH0;
const StyledH1 = /*#__PURE__*/_styledComponents.default.h1.withConfig({
  displayName: "styled__StyledH1",
  componentId: "sc-tqnj75-1"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.typography.fontSize.xxLarge;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.typography.fontWeight.light;
}, _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.lineHeight.xxLarge;
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.spacing.xLarge;
}, (0, _mixins.withMargins)());
exports.StyledH1 = StyledH1;
const StyledH2 = /*#__PURE__*/_styledComponents.default.h2.withConfig({
  displayName: "styled__StyledH2",
  componentId: "sc-tqnj75-2"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";", ";"], props => commonTextStyles(props), _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.typography.fontSize.xLarge;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.typography.fontWeight.regular;
}, _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.lineHeight.xLarge;
}, (0, _mixins.withMargins)());
exports.StyledH2 = StyledH2;
const StyledH3 = /*#__PURE__*/_styledComponents.default.h3.withConfig({
  displayName: "styled__StyledH3",
  componentId: "sc-tqnj75-3"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref15 => {
  let {
    theme
  } = _ref15;
  return theme.typography.fontSize.large;
}, _ref16 => {
  let {
    theme
  } = _ref16;
  return theme.typography.fontWeight.regular;
}, _ref17 => {
  let {
    theme
  } = _ref17;
  return theme.lineHeight.large;
}, _ref18 => {
  let {
    theme
  } = _ref18;
  return theme.spacing.small;
}, (0, _mixins.withMargins)());
exports.StyledH3 = StyledH3;
const StyledH4 = /*#__PURE__*/_styledComponents.default.h4.withConfig({
  displayName: "styled__StyledH4",
  componentId: "sc-tqnj75-4"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref19 => {
  let {
    theme
  } = _ref19;
  return theme.typography.fontSize.medium;
}, _ref20 => {
  let {
    theme
  } = _ref20;
  return theme.typography.fontWeight.semiBold;
}, _ref21 => {
  let {
    theme
  } = _ref21;
  return theme.lineHeight.medium;
}, _ref22 => {
  let {
    theme
  } = _ref22;
  return theme.spacing.xSmall;
}, (0, _mixins.withMargins)());
exports.StyledH4 = StyledH4;
const StyledText = /*#__PURE__*/_styledComponents.default.p.withConfig({
  displayName: "styled__StyledText",
  componentId: "sc-tqnj75-5"
})(["", " font-size:", ";font-weight:", ";line-height:", ";", " &:last-child{margin-bottom:0;}", ";"], props => commonTextStyles(props), _ref23 => {
  let {
    theme
  } = _ref23;
  return theme.typography.fontSize.medium;
}, _ref24 => {
  let {
    theme
  } = _ref24;
  return theme.typography.fontWeight.regular;
}, _ref25 => {
  let {
    theme
  } = _ref25;
  return theme.lineHeight.medium;
}, props => textModifiers(props), (0, _mixins.withMargins)());
exports.StyledText = StyledText;
const StyledSmall = /*#__PURE__*/_styledComponents.default.p.withConfig({
  displayName: "styled__StyledSmall",
  componentId: "sc-tqnj75-6"
})(["", ";color:", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", " &:last-child{margin-bottom:0;}", ";"], props => commonTextStyles(props), _ref26 => {
  let {
    color,
    theme
  } = _ref26;
  return color ? theme.colors[color] : theme.colors.secondary60;
}, _ref27 => {
  let {
    theme
  } = _ref27;
  return theme.typography.fontSize.small;
}, _ref28 => {
  let {
    theme
  } = _ref28;
  return theme.typography.fontWeight.regular;
}, _ref29 => {
  let {
    theme
  } = _ref29;
  return theme.lineHeight.small;
}, _ref30 => {
  let {
    theme
  } = _ref30;
  return theme.spacing.small;
}, props => textModifiers(props), (0, _mixins.withMargins)());
exports.StyledSmall = StyledSmall;
const StyledHR = /*#__PURE__*/_styledComponents.default.hr.withConfig({
  displayName: "styled__StyledHR",
  componentId: "sc-tqnj75-7"
})(["", ";border:0;border-bottom:1px solid ", ";"], (0, _mixins.withMargins)(), _ref31 => {
  let {
    color,
    theme
  } = _ref31;
  return color && color in theme.colors ? theme.colors[color] : theme.colors.secondary30;
});
exports.StyledHR = StyledHR;
StyledH0.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledH1.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledH2.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledH3.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledH4.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledText.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledSmall.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledHR.defaultProps = {
  theme: _bigDesignTheme.theme
};
