// @bigcommerce/big-design/dist/cjs/components/Typography/Typography.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Text = exports.StyleableText = exports.StyleableSmall = exports.StyleableHR = exports.StyleableH4 = exports.StyleableH3 = exports.StyleableH2 = exports.StyleableH1 = exports.StyleableH0 = exports.Small = exports.HR = exports.H4 = exports.H3 = exports.H2 = exports.H1 = exports.H0 = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const validHeadingTags = new Set(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']);

// Private
const StyleableText = _styled.StyledText;
exports.StyleableText = StyleableText;
const StyleableSmall = _styled.StyledSmall;
exports.StyleableSmall = StyleableSmall;
const StyleableH0 = _styled.StyledH0;
exports.StyleableH0 = StyleableH0;
const StyleableH1 = _styled.StyledH1;
exports.StyleableH1 = StyleableH1;
const StyleableH2 = _styled.StyledH2;
exports.StyleableH2 = StyleableH2;
const StyleableH3 = _styled.StyledH3;
exports.StyleableH3 = StyleableH3;
const StyleableH4 = _styled.StyledH4;
exports.StyleableH4 = StyleableH4;
const StyleableHR = _styled.StyledHR;

// Public
exports.StyleableHR = StyleableHR;
const Text = /*#__PURE__*/(0, _react.memo)(_ref => {
  let {
    className,
    style,
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(StyleableText, props);
});
exports.Text = Text;
const Small = /*#__PURE__*/(0, _react.memo)(_ref2 => {
  let {
    className,
    style,
    ...props
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(StyleableSmall, props);
});
exports.Small = Small;
const HR = /*#__PURE__*/(0, _react.memo)(_ref3 => {
  let {
    className,
    style,
    ...props
  } = _ref3;
  return /*#__PURE__*/_react.default.createElement(StyleableHR, props);
});
exports.HR = HR;
const H0 = /*#__PURE__*/(0, _react.memo)(_ref4 => {
  let {
    className,
    style,
    as,
    ...props
  } = _ref4;
  return /*#__PURE__*/_react.default.createElement(StyleableH0, (0, _extends2.default)({
    as: getHeadingTag('h1', as)
  }, props));
});
exports.H0 = H0;
const H1 = /*#__PURE__*/(0, _react.memo)(_ref5 => {
  let {
    className,
    style,
    as,
    ...props
  } = _ref5;
  return /*#__PURE__*/_react.default.createElement(StyleableH1, (0, _extends2.default)({
    as: getHeadingTag('h1', as)
  }, props));
});
exports.H1 = H1;
const H2 = /*#__PURE__*/(0, _react.memo)(_ref6 => {
  let {
    className,
    style,
    as,
    ...props
  } = _ref6;
  return /*#__PURE__*/_react.default.createElement(StyleableH2, (0, _extends2.default)({
    as: getHeadingTag('h2', as)
  }, props));
});
exports.H2 = H2;
const H3 = /*#__PURE__*/(0, _react.memo)(_ref7 => {
  let {
    className,
    style,
    as,
    ...props
  } = _ref7;
  return /*#__PURE__*/_react.default.createElement(StyleableH3, (0, _extends2.default)({
    as: getHeadingTag('h3', as)
  }, props));
});
exports.H3 = H3;
const H4 = /*#__PURE__*/(0, _react.memo)(_ref8 => {
  let {
    className,
    style,
    as,
    ...props
  } = _ref8;
  return /*#__PURE__*/_react.default.createElement(StyleableH4, (0, _extends2.default)({
    as: getHeadingTag('h4', as)
  }, props));
});
exports.H4 = H4;
const getHeadingTag = (defaultTag, tag) => {
  return tag && validHeadingTags.has(tag) ? tag : defaultTag;
};
Text.displayName = 'Text';
Small.displayName = 'Small';
HR.displayName = 'HR';
H0.displayName = 'H0';
H1.displayName = 'H1';
H2.displayName = 'H2';
H3.displayName = 'H3';
H4.displayName = 'H4';
