// @bigcommerce/big-design/dist/es/components/Table/SelectAll/SelectAll.js
import React from 'react';
import { Checkbox } from '../../Checkbox';
import { Flex, FlexItem } from '../../Flex';
import { Text } from '../../Typography';
export const SelectAll = _ref => {
  let {
    items = [],
    onChange,
    selectedItems,
    totalItems
  } = _ref;
  const allInPageSelected = items.length > 0 && items.every(item => selectedItems.has(item));
  const someInPageSelected = items.length > 0 && items.some(item => selectedItems.has(item));
  const handleSelectAll = () => {
    if (typeof onChange !== 'function') {
      return;
    }
    if (selectedItems.size === 0) {
      return onChange([...items]);
    }
    if (allInPageSelected) {
      const newSelectedItems = new Set(selectedItems);
      items.forEach(item => newSelectedItems.delete(item));
      return onChange([...newSelectedItems]);
    }
    return onChange([...new Set([...selectedItems, ...items])]);
  };
  if (typeof onChange !== 'function') {
    return null;
  }
  const totalSelectedItems = selectedItems.size;
  const label = allInPageSelected ? 'Deselect All' : 'Select All';
  return /*#__PURE__*/React.createElement(FlexItem, {
    flexShrink: 0,
    marginRight: "xxSmall"
  }, /*#__PURE__*/React.createElement(Flex, {
    flexDirection: "row"
  }, /*#__PURE__*/React.createElement(Checkbox, {
    checked: allInPageSelected,
    hiddenLabel: true,
    isIndeterminate: someInPageSelected,
    label: label,
    onChange: handleSelectAll
  }), /*#__PURE__*/React.createElement(Text, {
    marginLeft: "small"
  }, totalSelectedItems === 0 ? `${totalItems}` : `${totalSelectedItems}/${totalItems}`)));
};
