// @bigcommerce/big-design/dist/cjs/components/Worksheet/editors/ModalEditor/ModalEditor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalEditor = void 0;
var _react = _interopRequireWildcard(require("react"));
var _utils = require("../../../../utils");
var _Flex = require("../../../Flex");
var _Typography = require("../../../Typography");
var _hooks = require("../../hooks");
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const InternalModalEditor = _ref => {
  let {
    cell,
    formatting,
    isEditing
  } = _ref;
  const {
    store,
    useStore
  } = (0, _hooks.useWorksheetStore)();
  const setOpenModal = useStore(store, state => state.setOpenModal);
  const setEditingCell = useStore(store, state => state.setEditingCell);
  const {
    hash,
    value
  } = cell;
  const buttonRef = /*#__PURE__*/(0, _react.createRef)();
  (0, _react.useEffect)(() => {
    if (isEditing) {
      setOpenModal(hash);
    }
  }, [hash, isEditing, setOpenModal]);
  const handleClick = (0, _react.useCallback)(() => {
    setEditingCell({
      cell
    });
  }, [cell, setEditingCell]);
  const renderedValue = (0, _react.useMemo)(() => typeof formatting === 'function' ? formatting(value) : `${value}`, [formatting, value]);
  return /*#__PURE__*/_react.default.createElement(_Flex.Flex, {
    alignItems: "center",
    flexWrap: "wrap",
    justifyContent: "space-between"
  }, /*#__PURE__*/_react.default.createElement(_styled.StyledFlexItem, {
    flexShrink: 1,
    paddingRight: "small"
  }, /*#__PURE__*/_react.default.createElement(_Typography.Small, {
    color: cell.disabled ? 'secondary50' : 'secondary70',
    ellipsis: true,
    title: renderedValue
  }, renderedValue)), /*#__PURE__*/_react.default.createElement(_styled.StyledButton, {
    disabled: cell.disabled,
    onClick: handleClick,
    ref: buttonRef,
    variant: "subtle"
  }, "Edit"));
};
const ModalEditor = (0, _utils.typedMemo)(InternalModalEditor);
exports.ModalEditor = ModalEditor;
