// @bigcommerce/big-design-theme/dist/cjs/helpers/helpers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.emCalc = exports.createRGBA = exports.addValues = void 0;
Object.defineProperty(exports, "hideVisually", {
  enumerable: true,
  get: function () {
    return _polished.hideVisually;
  }
});
exports.remCalc = exports.listReset = void 0;
var _polished = require("polished");
var _styledComponents = require("styled-components");
var _options = require("../options");
const addValues = (first, second) => {
  return (0, _polished.math)(`${first} + ${second}`);
};

/**
 * Creates and rgba color giving a hex and an amount
 *
 * @param color hex color
 * @param alpha number between 0 and 1
 */
exports.addValues = addValues;
const createRGBA = (color, alpha) => {
  const calculatedAmount = 1 - alpha;
  return (0, _polished.transparentize)(calculatedAmount, color);
};
exports.createRGBA = createRGBA;
const remCalc = value => {
  const {
    htmlFontSize
  } = _options.themeOptions.getOptions();
  return (0, _polished.rem)(value, htmlFontSize);
};
exports.remCalc = remCalc;
const emCalc = value => {
  const {
    htmlFontSize
  } = _options.themeOptions.getOptions();
  return (0, _polished.em)(value, htmlFontSize);
};
exports.emCalc = emCalc;
const listReset = /*#__PURE__*/(0, _styledComponents.css)(["list-style-type:none;margin:0;padding:0;"]);
exports.listReset = listReset;
