// @bigcommerce/big-design/dist/cjs/components/PillTabs/PillTabs.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PillTabs = void 0;
var _bigDesignIcons = require("@bigcommerce/big-design-icons");
var _react = _interopRequireWildcard(require("react"));
var _hooks = require("../../hooks");
var _Button = require("../Button");
var _Dropdown = require("../Dropdown");
var _Flex = require("../Flex");
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const PillTabs = _ref => {
  let {
    activePills,
    items,
    onPillClick
  } = _ref;
  const parentRef = /*#__PURE__*/(0, _react.createRef)();
  const dropdownRef = /*#__PURE__*/(0, _react.createRef)();
  const [isMenuVisible, setIsMenuVisible] = (0, _react.useState)(false);
  const [pillsState, setPillsState] = (0, _react.useState)(items.map(item => ({
    isVisible: true,
    item,
    ref: /*#__PURE__*/(0, _react.createRef)()
  })));
  const hideOverflowedPills = (0, _react.useCallback)(() => {
    const parentWidth = parentRef.current?.offsetWidth;
    const dropdownWidth = dropdownRef.current?.offsetWidth;
    if (!parentWidth || !dropdownWidth) {
      return;
    }
    let remainingWidth = parentWidth;
    const newState = pillsState.map(stateObj => {
      const pillWidth = stateObj.ref.current?.offsetWidth;
      if (!pillWidth) {
        return stateObj;
      }
      if (remainingWidth - pillWidth > dropdownWidth) {
        remainingWidth -= pillWidth;
        return {
          ...stateObj,
          isVisible: true
        };
      }
      return {
        ...stateObj,
        isVisible: false
      };
    });
    const visiblePills = pillsState.filter(stateObj => stateObj.isVisible);
    const newVisiblePills = newState.filter(stateObj => stateObj.isVisible);
    if (visiblePills.length !== newVisiblePills.length) {
      setIsMenuVisible(newVisiblePills.length !== items.length);
      setPillsState(newState);
    }
  }, [items, parentRef, dropdownRef, pillsState]);
  const renderedDropdown = (0, _react.useMemo)(() => {
    const dropdownItems = pillsState.filter(stateObj => !stateObj.isVisible).map(stateObj => {
      const item = items.find(_ref2 => {
        let {
          title
        } = _ref2;
        return title === stateObj.item.title;
      });
      const isActive = item ? activePills.includes(item.id) : false;
      return {
        content: stateObj.item.title,
        onItemClick: () => onPillClick(stateObj.item.id),
        hash: stateObj.item.title.toLowerCase(),
        icon: isActive ? /*#__PURE__*/_react.default.createElement(_bigDesignIcons.CheckIcon, null) : undefined
      };
    });
    return /*#__PURE__*/_react.default.createElement(_styled.StyledFlexItem, {
      "data-testid": "pilltabs-dropdown-toggle",
      isVisible: isMenuVisible,
      ref: dropdownRef,
      role: "listitem"
    }, /*#__PURE__*/_react.default.createElement(_Dropdown.Dropdown, {
      items: dropdownItems,
      toggle: /*#__PURE__*/_react.default.createElement(_Button.Button, {
        iconOnly: /*#__PURE__*/_react.default.createElement(_bigDesignIcons.MoreHorizIcon, {
          title: "add"
        }),
        variant: "subtle"
      })
    }));
  }, [items, pillsState, isMenuVisible, dropdownRef, activePills, onPillClick]);
  (0, _react.useEffect)(() => {
    const itemIds = items.map(item => item.id);
    const stateIds = pillsState.map(stateItem => stateItem.item.id);

    // The item ids and their order must match exactly with the internal state, if not, the state needs to be synced up
    if (itemIds.join() !== stateIds.join()) {
      const newState = items.map(item => {
        const oldItem = pillsState.find(stateItem => stateItem.item === item);
        if (oldItem) {
          return oldItem;
        }
        return {
          // hideOverflownPills will correct this field if it needs correction
          isVisible: true,
          item,
          ref: /*#__PURE__*/(0, _react.createRef)()
        };
      });
      setPillsState(newState);
    }
  }, [items, pillsState]);
  const renderedPills = (0, _react.useMemo)(() => items.map((item, index) => {
    const pill = pillsState[index];
    if (!pill) {
      return;
    }
    return /*#__PURE__*/_react.default.createElement(_styled.StyledFlexItem, {
      "data-testid": `pilltabs-pill-${index}`,
      isVisible: pill.isVisible,
      key: index,
      ref: pill.ref,
      role: "listitem"
    }, /*#__PURE__*/_react.default.createElement(_styled.StyledPillTab, {
      disabled: !pill.isVisible,
      isActive: activePills.includes(item.id),
      marginRight: "xSmall",
      onClick: () => onPillClick(item.id),
      variant: "subtle"
    }, item.title));
  }), [items, pillsState, activePills, onPillClick]);
  (0, _react.useEffect)(() => {
    hideOverflowedPills();
  }, [items, parentRef, pillsState, hideOverflowedPills]);
  (0, _hooks.useWindowResizeListener)(() => {
    hideOverflowedPills();
  });
  return items.length > 0 ? /*#__PURE__*/_react.default.createElement(_Flex.Flex, {
    "data-testid": "pilltabs-wrapper",
    flexDirection: "row",
    flexWrap: "nowrap",
    ref: parentRef,
    role: "list"
  }, renderedPills, renderedDropdown) : null;
};
exports.PillTabs = PillTabs;
PillTabs.displayName = 'Pill Tabs';
