// @bigcommerce/big-design/dist/cjs/components/Worksheet/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Table = exports.StyledBox = exports.Header = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _Box = require("../Box");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const Table = /*#__PURE__*/_styledComponents.default.table.withConfig({
  displayName: "styled__Table",
  componentId: "sc-16rzzpq-0"
})(["border-collapse:collapse;border-spacing:0;min-width:", ";table-layout:fixed;width:", ";&:focus{outline:none;}", ""], _ref => {
  let {
    minWidth,
    hasStaticWidth
  } = _ref;
  return minWidth && !hasStaticWidth ? `${minWidth}px` : 'auto';
}, _ref2 => {
  let {
    hasStaticWidth
  } = _ref2;
  return hasStaticWidth ? 'auto' : '100%';
}, _ref3 => {
  let {
    hasExpandableRows
  } = _ref3;
  return hasExpandableRows && (0, _styledComponents.css)(["& > thead > tr{& > th:nth-of-type(2){padding-left:", ";}}& > tbody > tr{& > td:nth-of-type(3){padding-left:", ";}}"], _ref4 => {
    let {
      theme
    } = _ref4;
    return theme.spacing.xSmall;
  }, _ref5 => {
    let {
      theme
    } = _ref5;
    return theme.spacing.xSmall;
  });
});
exports.Table = Table;
Table.defaultProps = {
  theme: _bigDesignTheme.theme
};
const Header = /*#__PURE__*/_styledComponents.default.th.withConfig({
  displayName: "styled__Header",
  componentId: "sc-16rzzpq-1"
})(["border:", ";border-right-color:", ";box-sizing:border-box;color:", ";font-weight:", ";height:", ";overflow:hidden;padding:", ";text-align:", ";width:", ";"], _ref6 => {
  let {
    theme
  } = _ref6;
  return `${theme.helpers.remCalc(0.5)} solid ${theme.colors.secondary30}`;
}, _ref7 => {
  let {
    theme,
    columnType
  } = _ref7;
  return columnType === 'toggle' ? `${theme.colors.white}` : `${theme.colors.secondary30}`;
}, _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.colors.secondary60;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.typography.fontWeight.semiBold;
}, _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.helpers.remCalc(52);
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return `0 ${theme.helpers.remCalc(17)}`;
}, _ref12 => {
  let {
    columnType
  } = _ref12;
  if (columnType === 'number') {
    return 'right';
  }
  if (columnType === 'checkbox') {
    return 'center';
  }
  return 'left';
}, _ref13 => {
  let {
    columnWidth
  } = _ref13;
  return typeof columnWidth === 'string' ? columnWidth : `${columnWidth}px`;
});
exports.Header = Header;
const StyledBox = /*#__PURE__*/(0, _styledComponents.default)(_Box.Box).withConfig({
  displayName: "styled__StyledBox",
  componentId: "sc-16rzzpq-2"
})(["overflow-x:scroll;"]);
exports.StyledBox = StyledBox;
Header.defaultProps = {
  theme: _bigDesignTheme.theme
};
