// @bigcommerce/big-design/dist/cjs/components/Tree/TreeNode/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledText = exports.StyledSelectableWrapper = exports.StyledLi = exports.StyledGap = exports.StyledFlexItem = exports.StyledFlex = exports.StyledArrowWrapper = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _Flex = require("../../Flex");
var _private = require("../../Typography/private");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const StyledLi = /*#__PURE__*/_styledComponents.default.li.withConfig({
  displayName: "styled__StyledLi",
  componentId: "sc-1073t7q-0"
})(["outline:0;"]);
exports.StyledLi = StyledLi;
const StyledArrowWrapper = /*#__PURE__*/(0, _styledComponents.default)(_Flex.FlexItem).withConfig({
  displayName: "styled__StyledArrowWrapper",
  componentId: "sc-1073t7q-1"
})(["z-index:1;", ";"], _ref => {
  let {
    expanded
  } = _ref;
  return expanded && (0, _styledComponents.css)(["transform:rotate(90deg);"]);
});
exports.StyledArrowWrapper = StyledArrowWrapper;
const StyledSelectableWrapper = /*#__PURE__*/(0, _styledComponents.default)(_Flex.FlexItem).withConfig({
  displayName: "styled__StyledSelectableWrapper",
  componentId: "sc-1073t7q-2"
})(["padding:1px;z-index:1;"]);
exports.StyledSelectableWrapper = StyledSelectableWrapper;
const sharedAfterStyles = /*#__PURE__*/(0, _styledComponents.css)(["bottom:0;content:'';left:-1000px;right:0;position:absolute;top:0;z-index:0;"]);
const StyledFlex = /*#__PURE__*/(0, _styledComponents.default)(_Flex.Flex).withConfig({
  displayName: "styled__StyledFlex",
  componentId: "sc-1073t7q-3"
})(["cursor:pointer;min-height:", ";position:relative;li:focus > &,&:hover{&::after{", " background-color:", ";}}label,svg{vertical-align:middle;}", ""], _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.helpers.addValues(theme.spacing.xxLarge, theme.spacing.xSmall);
}, sharedAfterStyles, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.colors.secondary10;
}, _ref4 => {
  let {
    theme,
    selected
  } = _ref4;
  return selected && (0, _styledComponents.css)(["&::after{", " background-color:", ";}"], sharedAfterStyles, theme.colors.primary10);
});
exports.StyledFlex = StyledFlex;
const StyledFlexItem = /*#__PURE__*/(0, _styledComponents.default)(_Flex.FlexItem).withConfig({
  displayName: "styled__StyledFlexItem",
  componentId: "sc-1073t7q-4"
})(["z-index:1;"]);
exports.StyledFlexItem = StyledFlexItem;
const StyledGap = /*#__PURE__*/_styledComponents.default.div.withConfig({
  displayName: "styled__StyledGap",
  componentId: "sc-1073t7q-5"
})(["flex-shrink:0;height:", ";width:", ";"], _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.spacing.xLarge;
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.spacing.xLarge;
});
exports.StyledGap = StyledGap;
const StyledText = /*#__PURE__*/(0, _styledComponents.default)(_private.StyleableText).withConfig({
  displayName: "styled__StyledText",
  componentId: "sc-1073t7q-6"
})(["z-index:1;"]);
exports.StyledText = StyledText;
StyledArrowWrapper.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledFlex.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledGap.defaultProps = {
  theme: _bigDesignTheme.theme
};
