// @bigcommerce/big-design/dist/cjs/components/Stepper/Step/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledText = exports.StyledStep = exports.StyledSrOnlyText = exports.StyledLight = exports.StyledDash = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _polished = require("polished");
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _Box = require("../../Box");
var _Flex = require("../../Flex");
var _Grid = require("../../Grid");
var _private = require("../../Typography/private");
const StyledStep = /*#__PURE__*/(0, _styledComponents.default)(_Grid.Grid).withConfig({
  displayName: "styled__StyledStep",
  componentId: "sc-kstb4o-0"
})(["flex-basis:100%;gap:", ";grid-template-areas:'light dash' 'text text';grid-template-columns:auto 1fr;", "{grid-template-columns:", " 1fr;grid-template-rows:", " 1fr;max-width:", ";min-width:", ";}"], _ref => {
  let {
    theme
  } = _ref;
  return theme.spacing.xSmall;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.breakpoints.tablet;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.xLarge;
}, _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.spacing.xLarge;
}, _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.helpers.remCalc(228);
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.helpers.remCalc(96);
});
exports.StyledStep = StyledStep;
const StyledSrOnlyText = /*#__PURE__*/_styledComponents.default.span.withConfig({
  displayName: "styled__StyledSrOnlyText",
  componentId: "sc-kstb4o-1"
})(["", "{", "}"], _ref7 => {
  let {
    theme
  } = _ref7;
  return theme.breakpoints.tablet;
}, (0, _polished.hideVisually)());
exports.StyledSrOnlyText = StyledSrOnlyText;
const StyledLight = /*#__PURE__*/(0, _styledComponents.default)(_Flex.Flex).withConfig({
  displayName: "styled__StyledLight",
  componentId: "sc-kstb4o-2"
})(["border-radius:", ";color:", ";grid-area:light;user-select:none;", "{border-radius:", ";font-size:", ";line-height:", ";}"], _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.spacing.medium;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.colors.white;
}, _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.breakpoints.tablet;
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.borderRadius.circle;
}, _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.typography.fontSize.medium;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.lineHeight.medium;
});
exports.StyledLight = StyledLight;
const StyledDash = /*#__PURE__*/(0, _styledComponents.default)(_Box.Box).withConfig({
  displayName: "styled__StyledDash",
  componentId: "sc-kstb4o-3"
})(["align-self:center;height:", ";grid-area:dash;width:100%;"], _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.helpers.remCalc(2);
});
exports.StyledDash = StyledDash;
const StyledText = /*#__PURE__*/(0, _styledComponents.default)(_private.StyleableText).withConfig({
  displayName: "styled__StyledText",
  componentId: "sc-kstb4o-4"
})(["grid-area:text;"]);
exports.StyledText = StyledText;
StyledStep.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledSrOnlyText.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledLight.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledDash.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledText.defaultProps = {
  theme: _bigDesignTheme.theme
};
