// @bigcommerce/big-design/dist/cjs/components/Worksheet/hooks/useWorksheetStore/useWorksheetStore.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useWorksheetStore = exports.createWorksheetStore = void 0;
var _react = require("react");
var _zustand = require("zustand");
var _utils = require("../../utils");
var _Worksheet = require("../../Worksheet");
const createWorksheetStore = () => (0, _zustand.createStore)(set => ({
  columns: [],
  editedCells: [],
  editedCellsMap: {},
  editingCell: null,
  isMetaKey: false,
  isShiftPressed: false,
  isAutoFillActive: false,
  isSelectingActive: false,
  isBlockedFillOut: false,
  isControlKey: false,
  editWithValue: '',
  expandableRows: {},
  disabledRows: [],
  hiddenRows: [],
  invalidCells: [],
  invalidCellsMap: {},
  openedModal: null,
  rows: [],
  selectedCells: [],
  selectedCellsMap: {},
  selectedRows: [],
  tableRef: null,
  addEditedCells: cells => set(state => {
    const editedCells = (0, _utils.mergeCells)(state.editedCells, cells);
    return {
      ...state,
      editedCells,
      editedCellsMap: (0, _utils.getCellsMap)(editedCells)
    };
  }),
  addInvalidCells: cells => set(state => {
    const invalidCells = (0, _utils.mergeCells)(state.invalidCells, cells);
    return {
      ...state,
      invalidCells,
      invalidCellsMap: (0, _utils.getCellsMap)(invalidCells)
    };
  }),
  removeInvalidCells: cells => set(state => {
    const invalidCells = (0, _utils.deleteCells)(state.invalidCells, cells);
    return {
      ...state,
      invalidCells,
      invalidCellsMap: (0, _utils.getCellsMap)(invalidCells)
    };
  }),
  resetInvalidCells: () => set(state => ({
    ...state,
    invalidCells: [],
    invalidCellsMap: {}
  })),
  setColumns: columns => set(state => ({
    ...state,
    columns
  })),
  setExpandableRows: (expandableRows, defaultExpandedRows) => set(state => ({
    ...state,
    expandableRows,
    hiddenRows: (0, _utils.getHiddenRows)(expandableRows, defaultExpandedRows)
  })),
  setEditingCell: _ref => {
    let {
      cell,
      isMetaKey = false,
      isControlKey = false,
      editWithValue = ''
    } = _ref;
    set(state => {
      return {
        ...state,
        editingCell: cell,
        isMetaKey,
        isControlKey,
        editWithValue
      };
    });
  },
  setShiftPressed: isShiftPressed => set(state => ({
    ...state,
    isShiftPressed
  })),
  setDisabledRows: disabledRows => set(state => ({
    ...state,
    disabledRows
  })),
  setHiddenRows: hiddenRows => set(state => ({
    ...state,
    hiddenRows
  })),
  setOpenModal: value => set(state => ({
    ...state,
    openedModal: value
  })),
  setRows: rows => set(state => ({
    ...state,
    rows
  })),
  setSelectedCells: cells => set(state => ({
    ...state,
    selectedCells: cells,
    selectedCellsMap: (0, _utils.getCellsMap)(cells)
  })),
  setSelectedRows: rowIndexes => set(state => ({
    ...state,
    selectedRows: rowIndexes
  })),
  setTableRef: ref => set(state => ({
    ...state,
    tableRef: ref
  })),
  setAutoFillActive: isActive => set(state => ({
    ...state,
    isAutoFillActive: isActive
  })),
  setSelectingActive: isActive => set(state => ({
    ...state,
    isSelectingActive: isActive
  })),
  setBlockFillOut: isBlock => set(state => ({
    ...state,
    isBlockedFillOut: isBlock
  }))
}));
exports.createWorksheetStore = createWorksheetStore;
const useWorksheetStore = () => {
  const store = (0, _react.useContext)(_Worksheet.WorksheetContext);
  if (!store) {
    throw new Error('Worksheet store value is null');
  }
  return {
    store,
    useStore: _zustand.useStore
  };
};
exports.useWorksheetStore = useWorksheetStore;
