// @bigcommerce/big-design/dist/es/components/Chip/Chip.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { CloseIcon } from '@bigcommerce/big-design-icons';
import React, { memo } from 'react';
import { Text } from '../Typography';
import { StyledChip, StyledCloseButton } from './styled';
export const Chip = /*#__PURE__*/memo(_ref => {
  let {
    children,
    label,
    onDelete,
    ...rest
  } = _ref;
  const ariaLabel = label ? {
    'aria-label': `Remove ${label}`
  } : {};
  const renderDeleteButton = () => onDelete && /*#__PURE__*/React.createElement(StyledCloseButton, _extends({}, ariaLabel, {
    iconOnly: /*#__PURE__*/React.createElement(CloseIcon, {
      size: "medium",
      title: "Delete"
    }),
    onClick: onDelete,
    type: "button",
    variant: "subtle"
  }));
  return /*#__PURE__*/React.createElement(StyledChip, _extends({
    margin: "xxSmall"
  }, rest, {
    backgroundColor: "secondary30",
    borderRadius: "normal",
    paddingLeft: "xSmall",
    paddingRight: "xxSmall"
  }), /*#__PURE__*/React.createElement(Text, {
    margin: "none",
    marginRight: "xxSmall"
  }, label), renderDeleteButton());
});
Chip.displayName = 'Chip';
