// @bigcommerce/big-design/dist/cjs/components/TableNext/SelectAll/SelectAll.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectAll = void 0;
var _react = _interopRequireDefault(require("react"));
var _Checkbox = require("../../Checkbox");
var _Flex = require("../../Flex");
var _Typography = require("../../Typography");
var _useSelectAllState = require("./useSelectAllState");
const SelectAll = props => {
  const {
    allInPageSelected,
    handleSelectAll,
    label,
    someInPageSelected,
    totalSelectedItems
  } = (0, _useSelectAllState.useSelectAllState)(props);
  const {
    totalItems
  } = props;
  return /*#__PURE__*/_react.default.createElement(_Flex.FlexItem, {
    flexShrink: 0,
    marginRight: "xxSmall"
  }, /*#__PURE__*/_react.default.createElement(_Flex.Flex, {
    flexDirection: "row"
  }, /*#__PURE__*/_react.default.createElement(_Checkbox.Checkbox, {
    checked: allInPageSelected,
    hiddenLabel: true,
    isIndeterminate: someInPageSelected,
    label: label,
    onChange: handleSelectAll
  }), /*#__PURE__*/_react.default.createElement(_Typography.Text, {
    marginLeft: "small"
  }, totalSelectedItems === 0 ? `${totalItems}` : `${totalSelectedItems}/${totalItems}`)));
};
exports.SelectAll = SelectAll;
