// @bigcommerce/big-design/dist/es/components/Form/Group/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';
import { Flex } from '../../Flex';
const SharedGroupStyles = /*#__PURE__*/css(["display:grid;grid-gap:", ";margin-bottom:", ";&:last-child{margin-bottom:", ";}"], _ref => {
  let {
    theme
  } = _ref;
  return `${theme.spacing.xSmall} ${theme.spacing.medium}`;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.spacing.medium;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.none;
});
export const StyledError = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styled__StyledError",
  componentId: "sc-1a6yr6l-0"
})(["flex-direction:row;"]);
export const StyledGroup = /*#__PURE__*/styled.div.withConfig({
  displayName: "styled__StyledGroup",
  componentId: "sc-1a6yr6l-1"
})(["", ";"], SharedGroupStyles);
export const StyledInlineGroup = /*#__PURE__*/styled.div.withConfig({
  displayName: "styled__StyledInlineGroup",
  componentId: "sc-1a6yr6l-2"
})(["", ";", "{", " ", "}"], SharedGroupStyles, _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.breakpoints.tablet;
}, _ref5 => {
  let {
    childrenCount,
    fullWidth,
    theme
  } = _ref5;
  return childrenCount === 2 && css(["grid-template-columns:repeat(2,", ");", "{grid-column:1 / 3;}"], fullWidth ? '1fr' : theme.helpers.remCalc(200), StyledError);
}, _ref6 => {
  let {
    childrenCount,
    fullWidth,
    theme
  } = _ref6;
  return childrenCount === 3 && css(["grid-template-columns:repeat(3,", ");", "{grid-column:1 / 4;}"], fullWidth ? '1fr' : theme.helpers.remCalc(128), StyledError);
});
StyledError.defaultProps = {
  theme: defaultTheme
};
StyledGroup.defaultProps = {
  theme: defaultTheme
};
StyledInlineGroup.defaultProps = {
  theme: defaultTheme
};
