// @bigcommerce/big-design/dist/es/components/InlineMessage/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';
import { getBorderStyle } from '../../utils';
import { Flex } from '../Flex';
import { Grid } from '../Grid';
import { Link } from '../Link';
import { StyleableH4, StyleableSmall } from '../Typography/private';
export const StyledInlineMessage = /*#__PURE__*/styled(Grid).withConfig({
  displayName: "styled__StyledInlineMessage",
  componentId: "sc-14477sa-0"
})(["border:", ";border-radius:", ";grid-gap:", ";grid-template-columns:", ";padding:", ";", " ", ";"], _ref => {
  let {
    theme
  } = _ref;
  return theme.border.box;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.borderRadius.normal;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.xSmall;
}, _ref4 => {
  let {
    theme
  } = _ref4;
  return `${theme.spacing.large} 1fr ${theme.spacing.large}`;
}, _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.spacing.xSmall;
}, _ref6 => {
  let {
    onClose
  } = _ref6;
  return onClose ? css(["grid-template-areas:'icon messages close';grid-template-columns:", ";"], _ref7 => {
    let {
      theme
    } = _ref7;
    return `${theme.spacing.large} 1fr ${theme.spacing.medium}`;
  }) : css(["grid-template-areas:'icon messages';grid-template-columns:", ";"], _ref8 => {
    let {
      theme
    } = _ref8;
    return `${theme.spacing.large} 1fr`;
  });
}, _ref9 => {
  let {
    theme,
    type
  } = _ref9;
  return type && getBorderStyle(type, theme);
});
export const StyledHeader = /*#__PURE__*/styled(StyleableH4).withConfig({
  displayName: "styled__StyledHeader",
  componentId: "sc-14477sa-1"
})(["font-size:", ";line-height:", ";margin-bottom:", ";"], _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.typography.fontSize.small;
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.spacing.large;
}, _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.spacing.none;
});
export const StyledMessageItem = /*#__PURE__*/styled(StyleableSmall).attrs({
  as: 'span'
}).withConfig({
  displayName: "styled__StyledMessageItem",
  componentId: "sc-14477sa-2"
})(["color:", ";vertical-align:middle;"], _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.colors.secondary70;
});
export const StyledLink = /*#__PURE__*/styled(Link).withConfig({
  displayName: "styled__StyledLink",
  componentId: "sc-14477sa-3"
})(["font-size:", ";vertical-align:middle;"], _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.typography.fontSize.small;
});
export const StyledActionsWrapper = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styled__StyledActionsWrapper",
  componentId: "sc-14477sa-4"
})(["margin-bottom:-", ";margin-left:-", ";margin-right:-", ";"], _ref15 => {
  let {
    theme
  } = _ref15;
  return theme.spacing.xSmall;
}, _ref16 => {
  let {
    theme
  } = _ref16;
  return theme.spacing.xxSmall;
}, _ref17 => {
  let {
    theme
  } = _ref17;
  return theme.spacing.xxSmall;
});
StyledInlineMessage.defaultProps = {
  theme: defaultTheme
};
StyledHeader.defaultProps = {
  theme: defaultTheme
};
StyledMessageItem.defaultProps = {
  theme: defaultTheme
};
StyledLink.defaultProps = {
  theme: defaultTheme
};
StyledActionsWrapper.defaultProps = {
  theme: defaultTheme
};
