// @bigcommerce/big-design/dist/cjs/components/Form/Group/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledInlineGroup = exports.StyledGroup = exports.StyledError = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _Flex = require("../../Flex");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const SharedGroupStyles = /*#__PURE__*/(0, _styledComponents.css)(["display:grid;grid-gap:", ";margin-bottom:", ";&:last-child{margin-bottom:", ";}"], _ref => {
  let {
    theme
  } = _ref;
  return `${theme.spacing.xSmall} ${theme.spacing.medium}`;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.spacing.medium;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.none;
});
const StyledError = /*#__PURE__*/(0, _styledComponents.default)(_Flex.Flex).withConfig({
  displayName: "styled__StyledError",
  componentId: "sc-1a6yr6l-0"
})(["flex-direction:row;"]);
exports.StyledError = StyledError;
const StyledGroup = /*#__PURE__*/_styledComponents.default.div.withConfig({
  displayName: "styled__StyledGroup",
  componentId: "sc-1a6yr6l-1"
})(["", ";"], SharedGroupStyles);
exports.StyledGroup = StyledGroup;
const StyledInlineGroup = /*#__PURE__*/_styledComponents.default.div.withConfig({
  displayName: "styled__StyledInlineGroup",
  componentId: "sc-1a6yr6l-2"
})(["", ";", "{", " ", "}"], SharedGroupStyles, _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.breakpoints.tablet;
}, _ref5 => {
  let {
    childrenCount,
    fullWidth,
    theme
  } = _ref5;
  return childrenCount === 2 && (0, _styledComponents.css)(["grid-template-columns:repeat(2,", ");", "{grid-column:1 / 3;}"], fullWidth ? '1fr' : theme.helpers.remCalc(200), StyledError);
}, _ref6 => {
  let {
    childrenCount,
    fullWidth,
    theme
  } = _ref6;
  return childrenCount === 3 && (0, _styledComponents.css)(["grid-template-columns:repeat(3,", ");", "{grid-column:1 / 4;}"], fullWidth ? '1fr' : theme.helpers.remCalc(128), StyledError);
});
exports.StyledInlineGroup = StyledInlineGroup;
StyledError.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledGroup.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledInlineGroup.defaultProps = {
  theme: _bigDesignTheme.theme
};
