// @bigcommerce/big-design/dist/cjs/components/List/Item/Item.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListItem = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _bigDesignIcons = require("@bigcommerce/big-design-icons");
var _react = _interopRequireWildcard(require("react"));
var _utils = require("../../../utils");
var _Checkbox = require("../../Checkbox");
var _Content = require("./Content");
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const StyleableListItem = (0, _utils.typedMemo)(_ref => {
  let {
    actionType = 'normal',
    autoWidth = false,
    forwardedRef,
    index,
    isAction = false,
    isChecked = false,
    isHighlighted,
    isSelected = false,
    item,
    getItemProps,
    addItem,
    removeItem,
    ...props
  } = _ref;
  return removeItem && addItem ? /*#__PURE__*/_react.default.createElement(_styled.StyledListItem, (0, _extends2.default)({}, getItemProps({
    ...props,
    disabled: item.disabled,
    index,
    item,
    onClick: () => {
      if (item.disabled) {
        return;
      }
      const hasValue = item => 'value' in item;
      if (hasValue(item)) {
        isChecked ? removeItem(item) : addItem(item);
      }
    },
    ref: forwardedRef
  }), {
    actionType: actionType,
    autoWidth: autoWidth,
    isAction: isAction,
    isHighlighted: isHighlighted
  }), /*#__PURE__*/_react.default.createElement(_Checkbox.Checkbox, {
    checked: isChecked,
    description: item.description,
    disabled: item.disabled,
    label: item.content,
    onChange: () => null,
    onClick: event => {
      event.preventDefault();
      event.stopPropagation();
    },
    tabIndex: -1
  })) : /*#__PURE__*/_react.default.createElement(_styled.StyledListItem, (0, _extends2.default)({}, getItemProps({
    ...props,
    disabled: item.disabled,
    index,
    item,
    ref: forwardedRef
  }), {
    actionType: actionType,
    autoWidth: autoWidth,
    isAction: isAction,
    isHighlighted: isHighlighted,
    isSelected: isSelected
  }), /*#__PURE__*/_react.default.createElement(_Content.Content, {
    isHighlighted: isHighlighted,
    item: item
  }), isSelected && /*#__PURE__*/_react.default.createElement(_bigDesignIcons.CheckIcon, {
    color: "primary",
    size: "large"
  }));
});
const ListItem = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(StyleableListItem, (0, _extends2.default)({}, props, {
  forwardedRef: ref
}))));
exports.ListItem = ListItem;
