// @bigcommerce/big-design/dist/es/components/AccordionPanel/Accordion/Accordion.js
import { ExpandMoreIcon } from '@bigcommerce/big-design-icons';
import React, { memo, useId } from 'react';
import { StyledAccordionButton, StyledAccordionContent } from './styled';
export const Accordion = /*#__PURE__*/memo(_ref => {
  let {
    children,
    header,
    iconLeft,
    isExpanded,
    onClick
  } = _ref;
  const accordionId = useId();
  const accordionItemId = useId();
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(StyledAccordionButton, {
    "aria-controls": accordionItemId,
    "aria-expanded": isExpanded,
    iconLeft: iconLeft,
    iconRight: /*#__PURE__*/React.createElement(ExpandMoreIcon, {
      className: "collapse-icon",
      color: "secondary70"
    }),
    id: accordionId,
    isExpanded: isExpanded,
    onClick: onClick,
    type: "button",
    variant: "subtle"
  }, header), /*#__PURE__*/React.createElement(StyledAccordionContent, {
    "aria-labelledby": accordionId,
    display: isExpanded ? 'block' : 'none',
    hidden: !isExpanded,
    iconLeft: iconLeft,
    id: accordionItemId,
    role: "region"
  }, children));
});
