// @bigcommerce/big-design/dist/es/components/Datepicker/Header.js
import { ChevronLeftIcon, ChevronRightIcon } from '@bigcommerce/big-design-icons';
import React from 'react';
import { Flex } from '..';
import { Button } from '../Button';
import { Text } from '../Typography';
const Header = _ref => {
  let {
    date = new Date(),
    decreaseMonth,
    increaseMonth,
    months,
    prevMonthButtonDisabled,
    nextMonthButtonDisabled
  } = _ref;
  return /*#__PURE__*/React.createElement(Flex, {
    alignItems: "center",
    justifyContent: "space-between"
  }, /*#__PURE__*/React.createElement(Button, {
    disabled: prevMonthButtonDisabled,
    iconOnly: /*#__PURE__*/React.createElement(ChevronLeftIcon, {
      title: "View previous month."
    }),
    onClick: decreaseMonth,
    title: "View previous month.",
    type: "button",
    variant: "subtle"
  }), /*#__PURE__*/React.createElement(Text, {
    as: "span",
    bold: true,
    marginBottom: "none"
  }, `${months[date.getMonth()]} ${date.getFullYear()}`), /*#__PURE__*/React.createElement(Button, {
    disabled: nextMonthButtonDisabled,
    iconOnly: /*#__PURE__*/React.createElement(ChevronRightIcon, {
      title: "View next month."
    }),
    onClick: increaseMonth,
    title: "View next month.",
    type: "button",
    variant: "subtle"
  }));
};
export default Header;
