// @bigcommerce/big-design/dist/cjs/components/ButtonGroup/ButtonGroup.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonGroup = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _bigDesignIcons = require("@bigcommerce/big-design-icons");
var _react = _interopRequireWildcard(require("react"));
var _hooks = require("../../hooks");
var _Dropdown = require("../Dropdown");
var _Flex = require("../Flex");
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const defaultLocalization = {
  more: 'more'
};
const excludeIconProps = _ref => {
  let {
    iconOnly,
    iconRight,
    iconLeft,
    ...actionProps
  } = _ref;
  return actionProps;
};
const ButtonGroup = /*#__PURE__*/(0, _react.memo)(_ref2 => {
  let {
    actions,
    localization = defaultLocalization,
    ...wrapperProps
  } = _ref2;
  const parentRef = /*#__PURE__*/(0, _react.createRef)();
  const dropdownRef = /*#__PURE__*/(0, _react.createRef)();
  const [isMenuVisible, setIsMenuVisible] = (0, _react.useState)(false);
  const [actionsState, setActionsState] = (0, _react.useState)([]);
  (0, _react.useEffect)(() => {
    setActionsState(actions.map(action => ({
      isVisible: true,
      action: excludeIconProps(action),
      ref: /*#__PURE__*/(0, _react.createRef)()
    })));
  }, [actions]);
  const hideOverflowedActions = (0, _react.useCallback)(() => {
    const parentWidth = parentRef.current?.offsetWidth ?? 0;
    const dropdownWidth = dropdownRef.current?.offsetWidth ?? 0;
    let remainingWidth = parentWidth;
    const nextState = actionsState.map(stateObj => {
      const actionWidth = stateObj.ref.current?.offsetWidth;
      if (!actionWidth) {
        return stateObj;
      }
      if (stateObj.action.actionType === 'destructive') {
        return {
          ...stateObj,
          isVisible: false
        };
      }
      if (remainingWidth - actionWidth > dropdownWidth) {
        remainingWidth -= actionWidth;
        return {
          ...stateObj,
          isVisible: true
        };
      }
      return {
        ...stateObj,
        isVisible: false
      };
    });
    const visibleActions = actionsState.filter(_ref3 => {
      let {
        isVisible
      } = _ref3;
      return isVisible;
    });
    const nextVisibleActions = nextState.filter(_ref4 => {
      let {
        isVisible
      } = _ref4;
      return isVisible;
    });
    if (visibleActions.length !== nextVisibleActions.length) {
      setActionsState(nextState);
    }
  }, [actionsState, dropdownRef, parentRef]);
  const renderedDropdown = (0, _react.useMemo)(() => /*#__PURE__*/_react.default.createElement(_styled.StyledFlexItem, {
    "data-testid": "buttongroup-dropdown",
    isVisible: isMenuVisible,
    ref: dropdownRef,
    role: "listitem"
  }, /*#__PURE__*/_react.default.createElement(_Dropdown.Dropdown, {
    items: actionsState.filter(_ref5 => {
      let {
        isVisible
      } = _ref5;
      return !isVisible;
    }).map(_ref6 => {
      let {
        action,
        ref
      } = _ref6;
      return {
        actionType: action.actionType,
        content: action.text,
        disabled: action.disabled,
        onItemClick: () => {
          if (ref.current) {
            ref.current.getElementsByTagName('button')[0].click();
          }
        },
        hash: action.text.toLowerCase(),
        icon: action.icon
      };
    }),
    placement: "bottom-end",
    toggle: /*#__PURE__*/_react.default.createElement(_styled.StyledButton, {
      borderRadius: actionsState.every(_ref7 => {
        let {
          isVisible
        } = _ref7;
        return !isVisible;
      }),
      iconOnly: /*#__PURE__*/_react.default.createElement(_bigDesignIcons.MoreHorizIcon, {
        title: localization.more
      }),
      type: "button",
      variant: "secondary"
    })
  })), [actionsState, dropdownRef, isMenuVisible, localization.more]);
  const renderedActions = (0, _react.useMemo)(() => [...actionsState].reverse().sort(_ref8 => {
    let {
      isVisible
    } = _ref8;
    return isVisible ? -1 : 1;
  }).map((_ref9, key) => {
    let {
      action,
      isVisible,
      ref
    } = _ref9;
    const {
      text,
      icon,
      ...buttonProps
    } = action;
    return /*#__PURE__*/_react.default.createElement(_styled.StyledFlexItem, {
      "data-testid": "buttongroup-item",
      isVisible: isVisible,
      key: key,
      ref: ref,
      role: "listitem"
    }, /*#__PURE__*/_react.default.createElement(_styled.StyledButton, (0, _extends2.default)({}, buttonProps, {
      variant: "secondary"
    }), text));
  }), [actionsState]);
  (0, _react.useEffect)(() => {
    const nextIsMenuVisible = actionsState.some(_ref10 => {
      let {
        isVisible
      } = _ref10;
      return !isVisible;
    });
    if (nextIsMenuVisible !== isMenuVisible) {
      setIsMenuVisible(nextIsMenuVisible);
    }
  }, [actionsState, isMenuVisible]);
  (0, _react.useEffect)(() => {
    hideOverflowedActions();
  }, [actions, parentRef, hideOverflowedActions]);
  (0, _hooks.useWindowResizeListener)(() => {
    hideOverflowedActions();
  });
  return actions.length > 0 ? /*#__PURE__*/_react.default.createElement(_Flex.Flex, (0, _extends2.default)({}, wrapperProps, {
    "data-testid": "buttongroup-wrapper",
    flexDirection: "row",
    flexWrap: "nowrap",
    ref: parentRef,
    role: "list"
  }), renderedActions, renderedDropdown) : null;
});
exports.ButtonGroup = ButtonGroup;
