// @bigcommerce/big-design/dist/es/components/Search/Search.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { SearchIcon } from '@bigcommerce/big-design-icons';
import React from 'react';
import { Button } from '../Button';
import { Flex, FlexItem } from '../Flex';
import { Form } from '../Form';
import { Input } from '../Input';
const defaultLocalization = {
  search: 'Search'
};
export const Search = _ref => {
  let {
    localization = defaultLocalization,
    value,
    onChange,
    onSubmit,
    placeholder,
    'aria-label': ariaLabel,
    autoComplete = 'off',
    ...props
  } = _ref;
  const handleSubmit = event => {
    event.preventDefault();
    event.stopPropagation();
    if (typeof onSubmit === 'function') {
      onSubmit(event);
    }
  };
  return /*#__PURE__*/React.createElement(Form, {
    fullWidth: true,
    onSubmit: handleSubmit
  }, /*#__PURE__*/React.createElement(Flex, {
    alignItems: "center",
    backgroundColor: "white",
    flexDirection: "row",
    paddingBottom: "xxSmall"
  }, /*#__PURE__*/React.createElement(FlexItem, {
    flexGrow: 1,
    marginRight: "small"
  }, /*#__PURE__*/React.createElement(Input, _extends({}, props, {
    "aria-label": ariaLabel ?? localization.search,
    autoComplete: autoComplete,
    iconLeft: /*#__PURE__*/React.createElement(SearchIcon, {
      color: "secondary50"
    }),
    onChange: onChange,
    placeholder: placeholder ?? localization.search,
    type: "search",
    value: value
  }))), /*#__PURE__*/React.createElement(Button, {
    mobileWidth: "auto",
    type: "submit",
    variant: "secondary"
  }, localization.search)));
};
