// @bigcommerce/big-design/dist/cjs/components/Input/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledInputWrapper = exports.StyledInputContent = exports.StyledInput = exports.StyledIconWrapper = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _mixins = require("../../mixins");
var _transitions = require("../../mixins/transitions");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const StyledInputWrapper = /*#__PURE__*/_styledComponents.default.span.withConfig({
  displayName: "styled__StyledInputWrapper",
  componentId: "sc-g32raa-0"
})(["", " align-items:center;background-color:", ";border-radius:", ";box-sizing:border-box;display:flex;min-height:", ";overflow:hidden;position:relative;width:100%;", ";&:hover:not([disabled]){", "}", ";&[disabled]{background-color:", ";}"], (0, _transitions.withTransition)(['border', 'box-shadow']), _ref => {
  let {
    theme
  } = _ref;
  return theme.colors.white;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.borderRadius.normal;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return (0, _bigDesignTheme.addValues)(theme.spacing.xxLarge, theme.spacing.xxSmall);
}, _ref4 => {
  let {
    error,
    theme
  } = _ref4;
  return (0, _styledComponents.css)(["border:", ";"], error ? theme.border.boxError : theme.border.box);
}, _ref5 => {
  let {
    error,
    theme
  } = _ref5;
  return error ? (0, _styledComponents.css)(["border:", ";"], theme.border.boxError) : (0, _styledComponents.css)(["border:1px solid ", ";"], theme.colors.secondary40);
}, _ref6 => {
  let {
    error,
    focus,
    theme
  } = _ref6;
  return focus && (0, _styledComponents.css)(["outline:none;box-shadow:0 0 0 4px ", ";"], error ? theme.colors.danger20 : theme.colors.primary20);
}, _ref7 => {
  let {
    theme
  } = _ref7;
  return theme.colors.secondary20;
});
exports.StyledInputWrapper = StyledInputWrapper;
const StyledInput = /*#__PURE__*/_styledComponents.default.input.withConfig({
  displayName: "styled__StyledInput",
  componentId: "sc-g32raa-1"
})(["background-color:inherit;border:0;box-sizing:border-box;color:", ";flex:1 1 40%;height:100%;padding:0;padding-left:", ";width:100%;&:focus{outline:none;}&::placeholder{color:", ";font-size:", ";}&:-webkit-autofill{background-color:", " !important;-webkit-box-shadow:0 0 0px 1000px ", " inset;}", ";", ";", ";", ";", ";&[disabled]{background-color:", ";}"], _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.colors.secondary70;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.spacing.xSmall;
}, _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.colors.secondary40;
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.typography.fontSize.medium;
}, _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.colors.primary10;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.colors.primary10;
}, _ref14 => {
  let {
    iconRight,
    theme
  } = _ref14;
  return iconRight && (0, _styledComponents.css)(["padding-right:", ";"], (0, _bigDesignTheme.addValues)(theme.spacing.xxSmall, theme.spacing.xxLarge));
}, _ref15 => {
  let {
    iconLeft,
    theme
  } = _ref15;
  return iconLeft && (0, _styledComponents.css)(["padding-left:", ";"], (0, _bigDesignTheme.addValues)(theme.spacing.xxSmall, theme.spacing.xxLarge));
}, _ref16 => {
  let {
    chips,
    theme
  } = _ref16;
  return chips && (0, _styledComponents.css)(["min-height:", ";padding-left:", ";padding-right:", ";"], theme.spacing.xLarge, theme.spacing.xxSmall, theme.spacing.none);
}, _ref17 => {
  let {
    chips,
    theme
  } = _ref17;
  return chips && chips.length && (0, _styledComponents.css)(["margin-top:", ";"], theme.spacing.xxSmall);
}, _ref18 => {
  let {
    chips
  } = _ref18;
  return !chips && (0, _styledComponents.css)(["min-height:", ";"], (0, _bigDesignTheme.remCalc)(34));
}, _ref19 => {
  let {
    theme
  } = _ref19;
  return theme.colors.secondary20;
});
exports.StyledInput = StyledInput;
const StyledIconWrapper = /*#__PURE__*/_styledComponents.default.div.withConfig({
  displayName: "styled__StyledIconWrapper",
  componentId: "sc-g32raa-2"
})(["align-items:center;color:", ";display:flex;height:100%;position:absolute;top:0;", " ", " ", ""], _ref20 => {
  let {
    theme
  } = _ref20;
  return theme.colors.secondary60;
}, (0, _mixins.withPaddings)(), _ref21 => {
  let {
    paddingLeft
  } = _ref21;
  return paddingLeft === 'xSmall' && (0, _styledComponents.css)(["left:0;"]);
}, _ref22 => {
  let {
    paddingRight
  } = _ref22;
  return paddingRight === 'xSmall' && (0, _styledComponents.css)(["right:0;"]);
});
exports.StyledIconWrapper = StyledIconWrapper;
const StyledInputContent = /*#__PURE__*/_styledComponents.default.div.withConfig({
  displayName: "styled__StyledInputContent",
  componentId: "sc-g32raa-3"
})(["align-items:center;box-sizing:border-box;display:flex;flex-wrap:wrap;flex:1;height:100%;", ";", ";"], _ref23 => {
  let {
    chips,
    theme
  } = _ref23;
  return chips && (0, _styledComponents.css)(["margin-left:", ";padding-right:", ";"], theme.spacing.xxSmall, (0, _bigDesignTheme.addValues)(theme.spacing.xxSmall, theme.spacing.xxLarge));
}, _ref24 => {
  let {
    chips,
    theme
  } = _ref24;
  return chips && chips.length && (0, _styledComponents.css)(["margin-bottom:", ";"], theme.spacing.xxSmall);
});
exports.StyledInputContent = StyledInputContent;
StyledInput.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledInputWrapper.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledIconWrapper.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledInputContent.defaultProps = {
  theme: _bigDesignTheme.theme
};
