// @bigcommerce/big-design/dist/cjs/components/Message/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledMessageItem = exports.StyledMessage = exports.StyledLink = exports.StyledHeader = exports.StyledActionsWrapper = void 0;
var _bigDesignTheme = require("@bigcommerce/big-design-theme");
var _styledComponents = _interopRequireWildcard(require("styled-components"));
var _utils = require("../../utils");
var _Flex = require("../Flex");
var _Grid = require("../Grid");
var _Link = require("../Link");
var _private = require("../Typography/private");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const StyledMessage = /*#__PURE__*/(0, _styledComponents.default)(_Grid.Grid).withConfig({
  displayName: "styled__StyledMessage",
  componentId: "sc-1fctd93-0"
})(["", " grid-gap:", ";padding:", ";", " ", ";"], _ref => {
  let {
    theme
  } = _ref;
  return theme.shadow.raised;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.spacing.small;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.small;
}, _ref4 => {
  let {
    onClose
  } = _ref4;
  return onClose ? (0, _styledComponents.css)(["grid-template-areas:'icon messages close';grid-template-columns:", ";"], _ref5 => {
    let {
      theme
    } = _ref5;
    return `${theme.spacing.xLarge} 1fr ${theme.spacing.large}`;
  }) : (0, _styledComponents.css)(["grid-template-areas:'icon messages';grid-template-columns:", ";"], _ref6 => {
    let {
      theme
    } = _ref6;
    return `${theme.spacing.xLarge} 1fr`;
  });
}, _ref7 => {
  let {
    theme,
    type
  } = _ref7;
  return type && (0, _utils.getBorderStyle)(type, theme);
});
exports.StyledMessage = StyledMessage;
const StyledHeader = /*#__PURE__*/(0, _styledComponents.default)(_private.StyleableH4).withConfig({
  displayName: "styled__StyledHeader",
  componentId: "sc-1fctd93-1"
})(["line-height:", ";margin-bottom:", ";"], _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.spacing.large;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.spacing.none;
});
exports.StyledHeader = StyledHeader;
const StyledMessageItem = /*#__PURE__*/(0, _styledComponents.default)(_private.StyleableSmall).attrs({
  as: 'span'
}).withConfig({
  displayName: "styled__StyledMessageItem",
  componentId: "sc-1fctd93-2"
})(["color:", ";vertical-align:middle;"], _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.colors.secondary70;
});
exports.StyledMessageItem = StyledMessageItem;
const StyledLink = /*#__PURE__*/(0, _styledComponents.default)(_Link.Link).withConfig({
  displayName: "styled__StyledLink",
  componentId: "sc-1fctd93-3"
})(["font-size:", ";vertical-align:middle;"], _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.typography.fontSize.small;
});
exports.StyledLink = StyledLink;
const StyledActionsWrapper = /*#__PURE__*/(0, _styledComponents.default)(_Flex.Flex).withConfig({
  displayName: "styled__StyledActionsWrapper",
  componentId: "sc-1fctd93-4"
})(["margin-bottom:-", ";margin-left:-", ";margin-right:-", ";"], _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.spacing.xSmall;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.spacing.xxSmall;
}, _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.spacing.xxSmall;
});
exports.StyledActionsWrapper = StyledActionsWrapper;
StyledMessage.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledHeader.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledMessageItem.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledLink.defaultProps = {
  theme: _bigDesignTheme.theme
};
StyledActionsWrapper.defaultProps = {
  theme: _bigDesignTheme.theme
};
