// @bigcommerce/big-design/dist/cjs/components/Button/Button.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyleableButton = exports.Button = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _react = _interopRequireWildcard(require("react"));
var _ProgressCircle = require("../ProgressCircle");
var _styled = require("./styled");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const LoadingSpinner = () => /*#__PURE__*/_react.default.createElement(_styled.LoadingSpinnerWrapper, {
  alignItems: "center"
}, /*#__PURE__*/_react.default.createElement(_ProgressCircle.ProgressCircle, {
  size: "xxSmall"
}));
const RawButton = /*#__PURE__*/(0, _react.memo)(_ref => {
  let {
    forwardedRef,
    isLoading,
    disabled,
    ...props
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_styled.StyledButton, (0, _extends2.default)({
    className: "bd-button"
  }, props, {
    disabled: isLoading || disabled,
    ref: forwardedRef
  }), isLoading ? /*#__PURE__*/_react.default.createElement(LoadingSpinner, null) : null, /*#__PURE__*/_react.default.createElement(_styled.ContentWrapper, {
    isLoading: isLoading
  }, !props.iconOnly && props.iconLeft, props.iconOnly, !props.iconOnly && props.children, !props.iconOnly && props.iconRight));
});
const StyleableButton = /*#__PURE__*/(0, _react.forwardRef)((props, ref) => /*#__PURE__*/_react.default.createElement(RawButton, (0, _extends2.default)({}, props, {
  forwardedRef: ref
})));
exports.StyleableButton = StyleableButton;
const Button = /*#__PURE__*/(0, _react.forwardRef)((_ref2, ref) => {
  let {
    className,
    style,
    ...props
  } = _ref2;
  return /*#__PURE__*/_react.default.createElement(RawButton, (0, _extends2.default)({}, props, {
    forwardedRef: ref
  }));
});
exports.Button = Button;
const defaultProps = {
  actionType: 'normal',
  isLoading: false,
  mobileWidth: '100%',
  variant: 'primary'
};
Button.displayName = 'Button';
Button.defaultProps = {
  ...defaultProps
};
StyleableButton.displayName = 'StyleableButton';
StyleableButton.defaultProps = {
  ...defaultProps
};
