// @bigcommerce/big-design/dist/es/components/Typography/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import { ellipsis } from 'polished';
import styled, { css } from 'styled-components';
import { withMargins } from '../../mixins';
const commonTextStyles = props => css(["color:", ";margin:0 0 ", ";", ";"], _ref => {
  let {
    theme
  } = _ref;
  return props.color ? theme.colors[props.color] : theme.colors.secondary70;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.spacing.medium;
}, props.ellipsis && ellipsis());
const textModifiers = props => css(["", " ", " ", " ", " ", " ", " ", ""], _ref3 => {
  let {
    theme
  } = _ref3;
  return props.bold && css(["font-weight:", ";"], theme.typography.fontWeight.semiBold);
}, () => props.italic && css(["font-style:italic;"]), () => props.underline && css(["text-decoration:underline;"]), () => props.strikethrough && css(["text-decoration:line-through;"]), () => props.capitalize && css(["text-transform:capitalize;"]), () => props.lowercase && css(["text-transform:lowercase;"]), () => props.uppercase && css(["text-transform:uppercase;"]));
export const StyledH0 = /*#__PURE__*/styled.h1.withConfig({
  displayName: "styled__StyledH0",
  componentId: "sc-tqnj75-0"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.typography.fontSize.xxxLarge;
}, _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.typography.fontWeight.extraLight;
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.lineHeight.xxxLarge;
}, _ref7 => {
  let {
    theme
  } = _ref7;
  return theme.spacing.xLarge;
}, withMargins());
export const StyledH1 = /*#__PURE__*/styled.h1.withConfig({
  displayName: "styled__StyledH1",
  componentId: "sc-tqnj75-1"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.typography.fontSize.xxLarge;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.typography.fontWeight.light;
}, _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.lineHeight.xxLarge;
}, _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.spacing.xLarge;
}, withMargins());
export const StyledH2 = /*#__PURE__*/styled.h2.withConfig({
  displayName: "styled__StyledH2",
  componentId: "sc-tqnj75-2"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";", ";"], props => commonTextStyles(props), _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.typography.fontSize.xLarge;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.typography.fontWeight.regular;
}, _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.lineHeight.xLarge;
}, withMargins());
export const StyledH3 = /*#__PURE__*/styled.h3.withConfig({
  displayName: "styled__StyledH3",
  componentId: "sc-tqnj75-3"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref15 => {
  let {
    theme
  } = _ref15;
  return theme.typography.fontSize.large;
}, _ref16 => {
  let {
    theme
  } = _ref16;
  return theme.typography.fontWeight.regular;
}, _ref17 => {
  let {
    theme
  } = _ref17;
  return theme.lineHeight.large;
}, _ref18 => {
  let {
    theme
  } = _ref18;
  return theme.spacing.small;
}, withMargins());
export const StyledH4 = /*#__PURE__*/styled.h4.withConfig({
  displayName: "styled__StyledH4",
  componentId: "sc-tqnj75-4"
})(["", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", ";"], props => commonTextStyles(props), _ref19 => {
  let {
    theme
  } = _ref19;
  return theme.typography.fontSize.medium;
}, _ref20 => {
  let {
    theme
  } = _ref20;
  return theme.typography.fontWeight.semiBold;
}, _ref21 => {
  let {
    theme
  } = _ref21;
  return theme.lineHeight.medium;
}, _ref22 => {
  let {
    theme
  } = _ref22;
  return theme.spacing.xSmall;
}, withMargins());
export const StyledText = /*#__PURE__*/styled.p.withConfig({
  displayName: "styled__StyledText",
  componentId: "sc-tqnj75-5"
})(["", " font-size:", ";font-weight:", ";line-height:", ";", " &:last-child{margin-bottom:0;}", ";"], props => commonTextStyles(props), _ref23 => {
  let {
    theme
  } = _ref23;
  return theme.typography.fontSize.medium;
}, _ref24 => {
  let {
    theme
  } = _ref24;
  return theme.typography.fontWeight.regular;
}, _ref25 => {
  let {
    theme
  } = _ref25;
  return theme.lineHeight.medium;
}, props => textModifiers(props), withMargins());
export const StyledSmall = /*#__PURE__*/styled.p.withConfig({
  displayName: "styled__StyledSmall",
  componentId: "sc-tqnj75-6"
})(["", ";color:", ";font-size:", ";font-weight:", ";line-height:", ";margin:0 0 ", ";", " &:last-child{margin-bottom:0;}", ";"], props => commonTextStyles(props), _ref26 => {
  let {
    color,
    theme
  } = _ref26;
  return color ? theme.colors[color] : theme.colors.secondary60;
}, _ref27 => {
  let {
    theme
  } = _ref27;
  return theme.typography.fontSize.small;
}, _ref28 => {
  let {
    theme
  } = _ref28;
  return theme.typography.fontWeight.regular;
}, _ref29 => {
  let {
    theme
  } = _ref29;
  return theme.lineHeight.small;
}, _ref30 => {
  let {
    theme
  } = _ref30;
  return theme.spacing.small;
}, props => textModifiers(props), withMargins());
export const StyledHR = /*#__PURE__*/styled.hr.withConfig({
  displayName: "styled__StyledHR",
  componentId: "sc-tqnj75-7"
})(["", ";border:0;border-bottom:1px solid ", ";"], withMargins(), _ref31 => {
  let {
    color,
    theme
  } = _ref31;
  return color && color in theme.colors ? theme.colors[color] : theme.colors.secondary30;
});
StyledH0.defaultProps = {
  theme: defaultTheme
};
StyledH1.defaultProps = {
  theme: defaultTheme
};
StyledH2.defaultProps = {
  theme: defaultTheme
};
StyledH3.defaultProps = {
  theme: defaultTheme
};
StyledH4.defaultProps = {
  theme: defaultTheme
};
StyledText.defaultProps = {
  theme: defaultTheme
};
StyledSmall.defaultProps = {
  theme: defaultTheme
};
StyledHR.defaultProps = {
  theme: defaultTheme
};
