// @bigcommerce/big-design/dist/es/components/Tree/TreeNode/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';
import { Flex, FlexItem } from '../../Flex';
import { StyleableText } from '../../Typography/private';
export const StyledLi = /*#__PURE__*/styled.li.withConfig({
  displayName: "styled__StyledLi",
  componentId: "sc-1073t7q-0"
})(["outline:0;"]);
export const StyledArrowWrapper = /*#__PURE__*/styled(FlexItem).withConfig({
  displayName: "styled__StyledArrowWrapper",
  componentId: "sc-1073t7q-1"
})(["z-index:1;", ";"], _ref => {
  let {
    expanded
  } = _ref;
  return expanded && css(["transform:rotate(90deg);"]);
});
export const StyledSelectableWrapper = /*#__PURE__*/styled(FlexItem).withConfig({
  displayName: "styled__StyledSelectableWrapper",
  componentId: "sc-1073t7q-2"
})(["padding:1px;z-index:1;"]);
const sharedAfterStyles = /*#__PURE__*/css(["bottom:0;content:'';left:-1000px;right:0;position:absolute;top:0;z-index:0;"]);
export const StyledFlex = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styled__StyledFlex",
  componentId: "sc-1073t7q-3"
})(["cursor:pointer;min-height:", ";position:relative;li:focus > &,&:hover{&::after{", " background-color:", ";}}label,svg{vertical-align:middle;}", ""], _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.helpers.addValues(theme.spacing.xxLarge, theme.spacing.xSmall);
}, sharedAfterStyles, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.colors.secondary10;
}, _ref4 => {
  let {
    theme,
    selected
  } = _ref4;
  return selected && css(["&::after{", " background-color:", ";}"], sharedAfterStyles, theme.colors.primary10);
});
export const StyledFlexItem = /*#__PURE__*/styled(FlexItem).withConfig({
  displayName: "styled__StyledFlexItem",
  componentId: "sc-1073t7q-4"
})(["z-index:1;"]);
export const StyledGap = /*#__PURE__*/styled.div.withConfig({
  displayName: "styled__StyledGap",
  componentId: "sc-1073t7q-5"
})(["flex-shrink:0;height:", ";width:", ";"], _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.spacing.xLarge;
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.spacing.xLarge;
});
export const StyledText = /*#__PURE__*/styled(StyleableText).withConfig({
  displayName: "styled__StyledText",
  componentId: "sc-1073t7q-6"
})(["z-index:1;"]);
StyledArrowWrapper.defaultProps = {
  theme: defaultTheme
};
StyledFlex.defaultProps = {
  theme: defaultTheme
};
StyledGap.defaultProps = {
  theme: defaultTheme
};
