// @bigcommerce/big-design/dist/es/components/Message/Message.js
import _extends from "@babel/runtime/helpers/esm/extends";
import { CloseIcon } from '@bigcommerce/big-design-icons';
import React, { memo, useMemo } from 'react';
import { excludeMarginProps, excludePaddingProps } from '../../mixins';
import { getActionVariant, getMessagingIcon } from '../../utils';
import { Box } from '../Box';
import { Button } from '../Button';
import { MessagingButton } from '../Button/private';
import { GridItem } from '../Grid';
import { StyledActionsWrapper, StyledHeader, StyledLink, StyledMessage, StyledMessageItem } from './styled';
const defaultLocalization = {
  close: 'Close'
};
export const Message = /*#__PURE__*/memo(_ref => {
  let {
    className,
    localization = defaultLocalization,
    style,
    header,
    ...props
  } = _ref;
  const filteredProps = excludePaddingProps(props);
  const icon = useMemo(() => props.type && getMessagingIcon(props.type), [props.type]);
  const renderedMessages = useMemo(() => props.messages.map((_ref2, index) => {
    let {
      text,
      link
    } = _ref2;
    return /*#__PURE__*/React.createElement(Box, {
      key: index
    }, /*#__PURE__*/React.createElement(StyledMessageItem, null, text), ' ', link && /*#__PURE__*/React.createElement(StyledLink, link, link.text));
  }), [props.messages]);
  const renderedHeader = useMemo(() => header && /*#__PURE__*/React.createElement(StyledHeader, null, header), [header]);
  const renderedActions = useMemo(() => props.actions && /*#__PURE__*/React.createElement(StyledActionsWrapper, {
    flexDirection: "row",
    flexWrap: "wrap",
    marginTop: "xSmall"
  }, props.actions.map((_ref3, index) => {
    let {
      text,
      variant = 'secondary',
      ...actionProps
    } = _ref3;
    return /*#__PURE__*/React.createElement(Button, _extends({}, excludeMarginProps(actionProps), {
      key: index,
      marginBottom: "xSmall",
      marginHorizontal: "xxSmall",
      mobileWidth: "auto",
      variant: getActionVariant(variant)
    }), text);
  })), [props.actions]);
  return /*#__PURE__*/React.createElement(StyledMessage, _extends({}, filteredProps, {
    backgroundColor: "white",
    role: "alert"
  }), /*#__PURE__*/React.createElement(GridItem, {
    gridArea: "icon"
  }, icon), /*#__PURE__*/React.createElement(GridItem, {
    gridArea: "messages"
  }, renderedHeader, renderedMessages, renderedActions), props.onClose && /*#__PURE__*/React.createElement(GridItem, null, /*#__PURE__*/React.createElement(MessagingButton, {
    iconOnly: /*#__PURE__*/React.createElement(CloseIcon, {
      size: "large",
      title: localization.close
    }),
    onClick: props.onClose
  })));
});
Message.defaultProps = {
  messages: [],
  type: 'success'
};
