// @bigcommerce/big-design/dist/es/components/Alert/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';
import { getBorderStyle } from '../../utils';
import { Grid } from '../Grid';
import { Link } from '../Link';
import { StyleableH4, StyleableSmall } from '../Typography/private';
export const StyledAlert = /*#__PURE__*/styled(Grid).withConfig({
  displayName: "styled__StyledAlert",
  componentId: "sc-136bu5l-0"
})(["", " animation:", " .5s ease-in-out;background-color:", ";grid-gap:", ";max-width:", ";padding:", ";position:fixed;right:", ";top:", ";z-index:", ";", " ", ";"], _ref => {
  let {
    theme
  } = _ref;
  return theme.shadow.floating;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.keyframes.fadeIn;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.colors.white;
}, _ref4 => {
  let {
    theme
  } = _ref4;
  return theme.spacing.small;
}, _ref5 => {
  let {
    theme
  } = _ref5;
  return theme.helpers.remCalc(456);
}, _ref6 => {
  let {
    theme
  } = _ref6;
  return theme.spacing.small;
}, _ref7 => {
  let {
    theme
  } = _ref7;
  return theme.spacing.medium;
}, _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.spacing.medium;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.zIndex.fixed;
}, _ref10 => {
  let {
    onClose
  } = _ref10;
  return onClose ? css(["grid-template-areas:'icon messages close';grid-template-columns:", ";"], _ref11 => {
    let {
      theme
    } = _ref11;
    return `${theme.spacing.xLarge} 1fr ${theme.spacing.large}`;
  }) : css(["grid-template-areas:'icon messages';grid-template-columns:", ";"], _ref12 => {
    let {
      theme
    } = _ref12;
    return `${theme.spacing.xLarge} 1fr`;
  });
}, _ref13 => {
  let {
    theme,
    type
  } = _ref13;
  return type && getBorderStyle(type, theme);
});
export const StyledHeader = /*#__PURE__*/styled(StyleableH4).withConfig({
  displayName: "styled__StyledHeader",
  componentId: "sc-136bu5l-1"
})(["line-height:", ";margin-bottom:", ";"], _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.spacing.medium;
}, _ref15 => {
  let {
    theme
  } = _ref15;
  return theme.spacing.xxSmall;
});
export const StyledMessageItem = /*#__PURE__*/styled(StyleableSmall).attrs({
  as: 'span'
}).withConfig({
  displayName: "styled__StyledMessageItem",
  componentId: "sc-136bu5l-2"
})(["color:", ";vertical-align:middle;"], _ref16 => {
  let {
    theme
  } = _ref16;
  return theme.colors.secondary70;
});
export const StyledLink = /*#__PURE__*/styled(Link).withConfig({
  displayName: "styled__StyledLink",
  componentId: "sc-136bu5l-3"
})(["font-size:", ";vertical-align:middle;"], _ref17 => {
  let {
    theme
  } = _ref17;
  return theme.typography.fontSize.small;
});
StyledAlert.defaultProps = {
  theme: defaultTheme
};
StyledHeader.defaultProps = {
  theme: defaultTheme
};
StyledMessageItem.defaultProps = {
  theme: defaultTheme
};
StyledLink.defaultProps = {
  theme: defaultTheme
};
