// @bigcommerce/big-design/dist/es/components/Message/styled.js
import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
import styled, { css } from 'styled-components';
import { getBorderStyle } from '../../utils';
import { Flex } from '../Flex';
import { Grid } from '../Grid';
import { Link } from '../Link';
import { StyleableH4, StyleableSmall } from '../Typography/private';
export const StyledMessage = /*#__PURE__*/styled(Grid).withConfig({
  displayName: "styled__StyledMessage",
  componentId: "sc-1fctd93-0"
})(["", " grid-gap:", ";padding:", ";", " ", ";"], _ref => {
  let {
    theme
  } = _ref;
  return theme.shadow.raised;
}, _ref2 => {
  let {
    theme
  } = _ref2;
  return theme.spacing.small;
}, _ref3 => {
  let {
    theme
  } = _ref3;
  return theme.spacing.small;
}, _ref4 => {
  let {
    onClose
  } = _ref4;
  return onClose ? css(["grid-template-areas:'icon messages close';grid-template-columns:", ";"], _ref5 => {
    let {
      theme
    } = _ref5;
    return `${theme.spacing.xLarge} 1fr ${theme.spacing.large}`;
  }) : css(["grid-template-areas:'icon messages';grid-template-columns:", ";"], _ref6 => {
    let {
      theme
    } = _ref6;
    return `${theme.spacing.xLarge} 1fr`;
  });
}, _ref7 => {
  let {
    theme,
    type
  } = _ref7;
  return type && getBorderStyle(type, theme);
});
export const StyledHeader = /*#__PURE__*/styled(StyleableH4).withConfig({
  displayName: "styled__StyledHeader",
  componentId: "sc-1fctd93-1"
})(["line-height:", ";margin-bottom:", ";"], _ref8 => {
  let {
    theme
  } = _ref8;
  return theme.spacing.large;
}, _ref9 => {
  let {
    theme
  } = _ref9;
  return theme.spacing.none;
});
export const StyledMessageItem = /*#__PURE__*/styled(StyleableSmall).attrs({
  as: 'span'
}).withConfig({
  displayName: "styled__StyledMessageItem",
  componentId: "sc-1fctd93-2"
})(["color:", ";vertical-align:middle;"], _ref10 => {
  let {
    theme
  } = _ref10;
  return theme.colors.secondary70;
});
export const StyledLink = /*#__PURE__*/styled(Link).withConfig({
  displayName: "styled__StyledLink",
  componentId: "sc-1fctd93-3"
})(["font-size:", ";vertical-align:middle;"], _ref11 => {
  let {
    theme
  } = _ref11;
  return theme.typography.fontSize.small;
});
export const StyledActionsWrapper = /*#__PURE__*/styled(Flex).withConfig({
  displayName: "styled__StyledActionsWrapper",
  componentId: "sc-1fctd93-4"
})(["margin-bottom:-", ";margin-left:-", ";margin-right:-", ";"], _ref12 => {
  let {
    theme
  } = _ref12;
  return theme.spacing.xSmall;
}, _ref13 => {
  let {
    theme
  } = _ref13;
  return theme.spacing.xxSmall;
}, _ref14 => {
  let {
    theme
  } = _ref14;
  return theme.spacing.xxSmall;
});
StyledMessage.defaultProps = {
  theme: defaultTheme
};
StyledHeader.defaultProps = {
  theme: defaultTheme
};
StyledMessageItem.defaultProps = {
  theme: defaultTheme
};
StyledLink.defaultProps = {
  theme: defaultTheme
};
StyledActionsWrapper.defaultProps = {
  theme: defaultTheme
};
