// @bigcommerce/big-design/dist/cjs/components/Search/Search.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Search = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _bigDesignIcons = require("@bigcommerce/big-design-icons");
var _react = _interopRequireDefault(require("react"));
var _Button = require("../Button");
var _Flex = require("../Flex");
var _Form = require("../Form");
var _Input = require("../Input");
const defaultLocalization = {
  search: 'Search'
};
const Search = _ref => {
  let {
    localization = defaultLocalization,
    value,
    onChange,
    onSubmit,
    placeholder,
    'aria-label': ariaLabel,
    autoComplete = 'off',
    ...props
  } = _ref;
  const handleSubmit = event => {
    event.preventDefault();
    event.stopPropagation();
    if (typeof onSubmit === 'function') {
      onSubmit(event);
    }
  };
  return /*#__PURE__*/_react.default.createElement(_Form.Form, {
    fullWidth: true,
    onSubmit: handleSubmit
  }, /*#__PURE__*/_react.default.createElement(_Flex.Flex, {
    alignItems: "center",
    backgroundColor: "white",
    flexDirection: "row",
    paddingBottom: "xxSmall"
  }, /*#__PURE__*/_react.default.createElement(_Flex.FlexItem, {
    flexGrow: 1,
    marginRight: "small"
  }, /*#__PURE__*/_react.default.createElement(_Input.Input, (0, _extends2.default)({}, props, {
    "aria-label": ariaLabel ?? localization.search,
    autoComplete: autoComplete,
    iconLeft: /*#__PURE__*/_react.default.createElement(_bigDesignIcons.SearchIcon, {
      color: "secondary50"
    }),
    onChange: onChange,
    placeholder: placeholder ?? localization.search,
    type: "search",
    value: value
  }))), /*#__PURE__*/_react.default.createElement(_Button.Button, {
    mobileWidth: "auto",
    type: "submit",
    variant: "secondary"
  }, localization.search)));
};
exports.Search = Search;
